ipld/read.go

```diff
@@ -88,6 +102,75 @@ func GetLeaf(ctx context.Context, dag ipld.NodeGetter, root cid.Cid, leaf, total
 	return GetLeaf(ctx, dag, root, leaf, total)
 }
 
+// GetProofsForShares fetches Merkle proofs for the given shares
+// and returns the result as an array of NamespacedShareWithProof.
+func GetProofsForShares(
+	ctx context.Context,
+	dag ipld.NodeGetter,
+	root cid.Cid,
+	shares [][]byte,
+) ([]*NamespacedShareWithProof, error) {
+	proofs := make([]*NamespacedShareWithProof, len(shares))
+
+	for index, share := range shares {
+		if share != nil {
+			proof := make([]cid.Cid, 0)
+			s, err := GetLeafData(ctx, root, uint32(index), uint32(len(shares)), dag)
+			if err != nil {
+				return nil, err
+			}
+			proof, err = GetProof(ctx, dag, root, proof, index, len(shares))
+			if err != nil {
+				return nil, err
+			}
```

Comment:
Pls make a TODO and issue, mentioning that `GetLeafData` and `GetProof` can be combined into one function, as they traverse the same tree until the same share. The diff is only the data they collect, which can be controlled with options in the proposed unified function.

Reply:
Added